db/schema_applier.hh

```diff
@@ -0,0 +1,100 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#pragma once
+
+#include "db/schema_tables.hh"
+#include "mutation/mutation.hh"
+#include "seastar/core/future.hh"
+#include "service/storage_proxy.hh"
+#include "query-result-set.hh"
+
+#include <seastar/core/distributed.hh>
+
+namespace db {
+
+namespace schema_tables {
+
+// Must be called on shard 0.
+future<semaphore_units<>> hold_merge_lock() noexcept;
```

Comment:
Does this need to be exported? I see it only used inside applier.

Reply:
yes, it's being used also in main.cc